src/software/ai/evaluation/calc_best_shot.cpp

```diff
@@ -95,6 +95,14 @@ std::optional<Shot> calcBestShotOnGoal(const Field &field, const Team &friendly_
                                        const std::vector<Robot> &robots_to_ignore,
                                        double radius)
 {
+    if (shot_origin.x() < field.friendlyGoalCenter().x() ||
+        shot_origin.x() > field.enemyGoalCenter().x() ||
+        shot_origin.y() > field.enemyCornerPos().y() ||
+        shot_origin.y() < field.enemyCornerNeg().y())
```

Comment:
Can't we just use `contain(field.fieldLines(), shot_origin)`?

Reply:
fieldLines() constructs a new Rectangle for every call which is pretty expensive relative to everything else. 

I was also thinking, should we only check x values? Is there a case (free kick for example) where the y value will be out of bounds?

Reply:
yeah x value should be fine. Please add a test for out of bounds x and out of bounds y and make sure we don't see the top angle bottom angle warning